### PR TITLE
cmd: add sql-bootstrap-data command

### PR DESCRIPTION
**Pull request by RaduBerinde**
#### bootstrap: rename constants

In this commit we rename the bootstrap data constants to more closely
match the filenames and remove the redundant comments.

Epic: none
Release note: None

#### cmd: add sql-boostrap-data command

This command generates bootstrap data files and variable declarations,
simplifying the process for moving to the next release version.

Informs: #112629
Release note: None

### CODE REVIEW

**Review by celiala — APPROVED** [review]
LGTM - nice cleanup!

**Comment by RaduBerinde** [comment]
TFTR!

bors r+

**Comment by andyyang890** [comment]
This look great! I created a follow-up PR adding a `dev` shorthand in https://github.com/cockroachdb/cockroach/pull/114182 so that you don't have to copy out a long `bazel` command.

File: pkg/sql/catalog/bootstrap/initial_values.go [hunk 1]
```diff
@@ -132,56 +132,30 @@ func (f hardCodedInitialValues) build(
...
-	//go:embed data/23_2_tenant.keys
-	nonSystem232 string
-
-	// nonSystem232SHA256 is the SHA-256 hash of the above constant.
-	//go:embed data/23_2_tenant.sha256
-	nonSystem232SHA256 string
-)
+// The following variables hold hardcoded bootstrap data for older versions (as
+// produced by the final release version). For each version, we have system and
+// non-system keys, and each set of keys has an associated SHA-256.
```
Comment (rafiss) [inline pkg/sql/catalog/bootstrap/initial_values.go:137]: it was kind of nice that the old comment said you how you should populate new values when the version changes. could there be instructions on how to do that with the new way?

Comment (RaduBerinde) [reply pkg/sql/catalog/bootstrap/initial_values.go:137]: @andyyang890 is adding a comment in #114120

I will also rewrite the runbooks for bumping versions and check them in as an `.md` file, and I will point to that here.

**Pull request merged**